File: Backtracking/Backtracking.py
class Backtracing:        
    def backtracking(self, current_rows, current_columns, path, destiny_row, destiny_column, check):
        if current_columns == destiny_column and current_rows == destiny_row:
            print(path)
            return
        
        if not check[current_rows][current_columns]:
            return
        
        check[current_rows][current_columns] = False

        # Down
        if current_rows < destiny_row:
            self.backtracking(current_rows+1,current_columns, path+"D", destiny_row, destiny_column,check)
        # Right
        if current_columns < destiny_column:
            self.backtracking(current_rows, current_columns+1, path+"R", destiny_row,destiny_column,check)
        # Up
        if current_rows > 0:
            self.backtracking(current_rows-1, current_columns, path+"U",destiny_row, destiny_column,check)
        # Left
        if current_columns > 0:
            self.backtracking(current_rows, current_columns-1, path+"L",destiny_row, destiny_column,check)

        check[current_rows][current_columns] = True

obj = Backtracing()
check = [
    [True, True, True],
    [True, True, True],
    [True, True, True],
    ]
obj.backtracking(0,0,"",2,2,check)

File: Backtracking/Backtracking.java
package Backtracking;

class Backtracking {

    private int maxRow;
    private int maxColumn;

    Backtracking(int r, int c) {
        this.maxRow = r;
        this.maxColumn = c;
    }

    void mazeproblem(int rows, int columns, String path, boolean[][] checkMaze) {
        if (rows == 1 && columns == 1) {
            System.out.println(path);
            return;
        }

        if(!checkMaze[this.maxRow-rows][this.maxColumn-columns]){
            return;
        }
        checkMaze[this.maxRow-rows][this.maxColumn-columns] = false;
        // Path to Down
        if (rows > 1) {
            mazeproblem(rows - 1, columns, path + "D",checkMaze);
        }
        // Path to Right
        if (columns > 1) {
            mazeproblem(rows, columns - 1, path + "R",checkMaze);
        }
        // Path to Up
        if (rows < this.maxRow ) {
            mazeproblem(rows + 1, columns, path + "U",checkMaze);
        }
        // Path to Left
        if (columns < this.maxColumn) {
            mazeproblem(rows, columns + 1, path + "L",checkMaze);
        }
        
        checkMaze[this.maxRow-rows][this.maxColumn-columns] = true;

    }

    public static void main(String[] args) {
        Backtracking obj = new Backtracking(3, 3);
        boolean[][] checkMaze = {
            {true, true, true},
            {true, true, true},
            {true, true, true},
        };
        obj.mazeproblem(3, 3, "",checkMaze);
    }
}
